Source/Common/Util.py

```diff
@@ -15,3 +15,73 @@ def scale_value(portion_of_screen, minimum_res, scale_value, maximum_res=None):
         return max(min(scaled_value, maximum_res), minimum_res)
     else:
         return max(scaled_value, minimum_res)
+
+
+class Vec2:
+    def __init__(self, x, y=None):
+        if isinstance(x, (tuple, list, Vec2)):
+            if len(x) != 2:
+                raise ValueError("Input tuple or list must have exactly 2 elements")
+            self.x, self.y = x
+        else:
+            if y == None:
+                raise ValueError(
+                    "Vec2() takes 2 values or 1 list/tuple but only 1 was value was given"
+                )
+            self.x = x
+            self.y = y
+
+    # funky way of making the operator stuff smaller
+    def selector(self, b, op):
+        return (
+            Vec2(op(self.x, b.x), op(self.y, b.y))
+            if type(b) == Vec2
+            else Vec2(op(self.x, b), op(self.y, b))
+        )
+
+    def __add__(self, b):
+        return self.selector(b, lambda x, y: x + y)
+
+    def __sub__(self, b):
+        return self.selector(b, lambda x, y: x - y)
+
+    def __mul__(self, b):
+        return self.selector(b, lambda x, y: x * y)
+
+    def __truediv__(self, b):
+        return self.selector(b, lambda x, y: x / y)
+
+    def __eq__(self, b):
+        return type(b) == Vec2 and self.x == b.x and self.y == b.y
+
+    # Allows for list like indexing, example Vec2(2,3)[0], returns 2
+    def __getitem__(self, index):
+        return self.y if index else self.x
+
+    def __str__(self):
+        return "[" + str(self.x) + "," + str(self.y) + "]"
+
+    def __len__(self):
+        return 2
+
+    def __iter__(self):
+        return iter((self.x, self.y))
+
+    def length(self):
+        return math.sqrt(self.x**2 + self.y**2)
+
+    def unit(self):
+        return self / math.sqrt(self.x**2 + self.y**2)
+
+
+def Collision(
+    pos, r, m
```

Comment:
We need to use more descriptive variable names, and we should have comments explaining what type a parameter should be.